ibis/file/csv/client.py

```diff
@@ -65,4 +66,26 @@ def list_databases(self, path=None):
 
 @data_preload.register(CSVTable, (pathlib.Path, str))
 def csv_data_preload_uri_client(table, path, scope=None, **kwargs):
-    return pd.read_csv(str(path), header=0)
+    if scope is not None:
+        usecols = scope.get('usecols')
+    else:
+        usecols = None
+    df = pd.read_csv(str(path), usecols=usecols, header=0)
+    return df
+
+
+@pre_execute.register(ops.Selection)
```

Comment:
I *really* want to register ``pre_execute`` on the ``ops.Selection(CSVTable)`` (rather than do this instance check). any way to do this?

Reply:
We can probably extend `pre_execute` to take a `client` argument, so that you don't have to do this instance checking. This would require a `_client` attribute on expressions or something like it.

Reply:
ok revised to just use ``pre_execute`` now.

but lots of hackiness because this is called pretty generally, IOW I only want to be called if I have a physical_table that is a CSVTable *in* the selection.

A tiny bit of hackiness with getting the path arg as well (because I can no longer dispatch on it), but that is ok.